Use convenience initializers over smart pointers when autoreleasing

LumioseSil opened the pull request:
#### 993dd98fc6dbf5ae452b1e7b87517867ebf5a70a
<pre>
Use convenience initializers over smart pointers when autoreleasing
<a href="https://bugs.webkit.org/show_bug.cgi?id=253518">https://bugs.webkit.org/show_bug.cgi?id=253518</a>

Reviewed by NOBODY (OOPS!).

Since we are autoreleasing anyway, having a smart pointer do the
redundant work is silly.

* Source/JavaScriptCore/API/JSManagedValue.mm:
* Source/JavaScriptCore/bytecode/Opcode.cpp:
* Source/WTF/wtf/cf/CFURLExtras.cpp:
* Source/WTF/wtf/cf/CFURLExtras.h:
* Source/WTF/wtf/cocoa/NSURLExtras.mm:
* Source/WebCore/Modules/applepay/cocoa/PaymentSummaryItemsCocoa.mm:
* Source/WebCore/PAL/pal/system/mac/PopupMenu.mm:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
* Source/WebCore/platform/LocalizedStrings.cpp:
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/platform/cocoa/SystemVersion.mm:
* Source/WebCore/platform/graphics/mac/controls/WebControlView.mm:
* Source/WebCore/platform/ios/WebItemProviderPasteboard.mm:
* Source/WebGPU/WebGPU/Adapter.mm:
* Source/WebGPU/WebGPU/Device.mm:
* Source/WebKit/Platform/cocoa/PaymentAuthorizationPresenter.mm:
* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
* Source/WebKit/Shared/ApplePay/cocoa/PaymentSetupConfiguration.mm:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
* Source/WebKit/Shared/mac/WebCoreArgumentCodersMac.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataRecord.mm:
* Source/WebKit/UIProcess/API/Cocoa/_WKTextManipulationToken.mm:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
* Source/WebKit/UIProcess/mac/WebPopupMenuProxyMac.mm:
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInNodeHandle.mm:
* Source/WebKitLegacy/mac/Misc/WebLocalizableStringsInternal.h:
* Source/WebKitLegacy/mac/Misc/WebLocalizableStringsInternal.mm:
* Source/WebKitLegacy/mac/Misc/WebNSURLExtras.mm:
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/993dd98fc6dbf5ae452b1e7b87517867ebf5a70a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1336 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/1375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1419 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/2239 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/1190 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1321 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1427 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/1435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/2239 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1348 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1255 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/1228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/2085 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/1268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/1213 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/2085 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/1169 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/1221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1248 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/2085 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/1304 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/1288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/1435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/1370 "Built successfully") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/1202 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/1220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/283 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1305 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/1374 "Built successfully") | 
| | | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/3/builds/278 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WTF/wtf/cocoa/NSURLExtras.mm:
```diff
@@ -115,7 +115,7 @@ static String decodePercentEscapes(const String& string)
     auto trimmed = adoptNS([string mutableCopy]);
     CFStringTrimWhitespace((__bridge CFMutableStringRef)trimmed.get());
-    return trimmed;
+    return adoptNS([trimmed copy]);
```
I don't think we want to do this. Why make a copy?

AZero13 replied:
We are returning an NSArray, not an NSMutableArray

hortont424 replied:
Does not matter; NSMutableArray is a NSArray subclass; the caller will never know, this is a needless copy.

rniwa commented on Source/WebCore/PAL/pal/system/mac/PopupMenu.mm:
```diff
@@ -53,20 +53,30 @@ void popUpMenu(NSMenu *menu, NSPoint location, float width, NSView *view, int se
+    NSSize labelOffset;
     if (menu.userInterfaceLayoutDirection == NSUserInterfaceLayoutDirectionRightToLeft)
         labelOffset = NSMakeSize(24, 1);
+    else
```
This seems strictly worse.

AZero13 replied:
I think avoiding an extra construction is worth it, unless the compiler knows how to optimize that out as it was before.

hortont424 reviewed (changes requested):
Please undo all the needless copies

cdumez commented on Source/WebKitLegacy/mac/WebView/WebResource.mm:
```diff
@@ -338,12 +338,12 @@ - (NSString *)_suggestedFilename
         filename = [[self URL] _webkit_suggestedFilenameWithMIMEType:[self MIMEType]];
     [wrapper setPreferredFilename:filename];
-    return wrapper.autorelease();
+    return adoptNS(wrapper).autorelease();
```
I disagree with some of the changes in this patch. We should adopt as early as possible.

AZero13 replied:
Okay then. Reverting those changes.

LumioseSil committed:
Use convenience initializers over smart pointers when autoreleasing
https://bugs.webkit.org/show_bug.cgi?id=253518

Reviewed by NOBODY (OOPS!).

Since we are autoreleasing anyway, having a smart pointer do the
redundant work is silly.

* Source/JavaScriptCore/API/JSManagedValue.mm:
* Source/JavaScriptCore/bytecode/Opcode.cpp:
* Source/WTF/wtf/cf/CFURLExtras.cpp:
* Source/WTF/wtf/cf/CFURLExtras.h:
* Source/WTF/wtf/cocoa/NSURLExtras.mm:
* Source/WebCore/Modules/applepay/cocoa/PaymentSummaryItemsCocoa.mm:
* Source/WebCore/PAL/pal/system/mac/PopupMenu.mm:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
* Source/WebCore/accessibility/mac/AXObjectCacheMac.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
* Source/WebCore/platform/LocalizedStrings.cpp:
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/platform/cocoa/SystemVersion.mm:
* Source/WebCore/platform/graphics/mac/controls/WebControlView.mm:
* Source/WebCore/platform/ios/WebItemProviderPasteboard.mm:
* Source/WebGPU/WebGPU/Adapter.mm:
* Source/WebGPU/WebGPU/Device.mm:
* Source/WebKit/Platform/cocoa/PaymentAuthorizationPresenter.mm:
* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
* Source/WebKit/Shared/ApplePay/cocoa/PaymentSetupConfiguration.mm:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
* Source/WebKit/Shared/mac/WebCoreArgumentCodersMac.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataRecord.mm:
* Source/WebKit/UIProcess/API/Cocoa/_WKTextManipulationToken.mm:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalAuthenticator.mm:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
* Source/WebKit/UIProcess/mac/WebPopupMenuProxyMac.mm:
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInNodeHandle.mm:
* Source/WebKitLegacy/mac/Misc/WebLocalizableStringsInternal.h:
* Source/WebKitLegacy/mac/Misc/WebLocalizableStringsInternal.mm:
* Source/WebKitLegacy/mac/Misc/WebNSURLExtras.mm:
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
